scaleway/resource_account_project_test.go

```diff
@@ -0,0 +1,161 @@
+package scaleway
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	accountV2 "github.com/scaleway/scaleway-sdk-go/api/account/v2"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func init() {
+	if !terraformBetaEnabled {
+		return
+	}
+	resource.AddTestSweepers("scaleway_account_project", &resource.Sweeper{
+		Name: "scaleway_account_project",
+		F:    testSweepAccountproject,
+	})
+}
+
+func testSweepAccountproject(_ string) error {
+	return sweep(func(scwClient *scw.Client) error {
+		accountAPI := accountV2.NewAPI(scwClient)
+
+		l.Debugf("sweeper: destroying the project")
+
+		listProjects, err := accountAPI.ListProjects(&accountV2.ListProjectsRequest{})
+		if err != nil {
+			return fmt.Errorf("failed to list projects: %w", err)
+		}
+		for _, project := range listProjects.Projects {
```

Comment:
Check if it is not the default project. The default project got the field `Name` at `default`. This name is not mutable. Otherwise, this code will break when it will try to delete the default project which is not deletable.